syncd/FlexCounter.cpp

```diff
@@ -1758,16 +1764,49 @@ void FlexCounter::endFlexCounterThread(void)
     SWSS_LOG_INFO("Flex Counter thread ended");
 }
 
-void FlexCounter::updateSupportedPortCounters(
-        _In_ sai_object_id_t portRid)
+sai_status_t FlexCounter::querySupportedPortCounters(
+        sai_object_id_t portRid)
 {
-    SWSS_LOG_ENTER();
+    sai_stat_capability_list_t stats_capability;
+    stats_capability.count = 0;
+    stats_capability.list = nullptr;
 
-    if (m_supportedPortCounters.size())
+    /* First call is to check the size needed to allocate */
+    sai_status_t status = m_vendorSai->queryStatsCapability(
+        portRid, 
+        SAI_OBJECT_TYPE_PORT, 
+        &stats_capability);
+
+    /* Second call is for query statistics capability */
+    if (status == SAI_STATUS_BUFFER_OVERFLOW)
     {
-        return;
+        std::vector<sai_stat_capability_t> statCapabilityList(stats_capability.count);
+        stats_capability.list = statCapabilityList.data();
+        status = m_vendorSai->queryStatsCapability(
+            portRid, 
+            SAI_OBJECT_TYPE_PORT, 
+            &stats_capability);
+        
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_INFO("Unable to get port supported counters for %s", 
+            sai_serialize_object_id(portRid).c_str());
+        }
+        else
+        {
+            for (uint32_t i = 0; i < stats_capability.count; i++)
+            {
+                sai_port_stat_t counter = static_cast<sai_port_stat_t>((stats_capability.list)[i].stat_enum);
```

Comment:
Why do you need extra parenthesis?
```suggestion
                sai_port_stat_t counter = static_cast<sai_port_stat_t>(stats_capability.list[i].stat_enum);
```

Reply:
Done